src/constructs/core/stage.test.ts

```diff
@@ -0,0 +1,101 @@
+import "@aws-cdk/assert/jest";
+
+import "../../utils/test/jest";
+import { SynthUtils } from "@aws-cdk/assert";
+import { App, CfnMapping } from "@aws-cdk/core";
+import { GuStack } from "./stack";
+import type { Stage } from "./stage";
+import { GuCertificateExample, NewStageMapping } from "./stage";
+
+describe("Stage refactor (POC)", () => {
+  it("should output a mapping as expected", () => {
+    const stack = new GuStack(new App(), "Test", { stack: "test" });
+
+    const mappings = NewStageMapping(new CfnMapping(stack, "foo"));
+    const domainName = mappings.set("domainName", stack.stage as Stage, {
+      CODE: "foo.example.com",
+      PROD: "prod.example.com",
+    });
+
+    new GuCertificateExample(stack, {
+      app: "foo",
+      domainName: domainName,
```

Comment:
(Construct just receives a string but actually it is a stage-dependant mapping ref.)